src/screens/view-wallet-mnemonic/view-wallet-mnemonic.tsx

```diff
@@ -0,0 +1,131 @@
+        this.state = {
+            mnemonic: new Array(24).fill(''),
+            showPasswordModal: true
+        };
+    }
+
+    public async onEnterPassword(password) {
+        const walletId = this.props.navigation.state?.params?.wallet.id;
+
+        if (!walletId) {
+            // TODO show an error?
+            this.props.navigation.goBack(null);
+        }
+
+        try {
+            const passHash = await hash(password);
+            const mnemonic = await readEncrypted(walletId, passHash);
+
+            this.setState({
+                showPasswordModal: false,
+                mnemonic: mnemonic.split(' ')
+            });
+        } catch (e) {
+            if (e.code === 'decrypt_fail') {
+                Alert.alert(translate('App.labels.error'), translate('Wallets.invalidPassword'), [
+                    { text: 'Ok' }
+                ]);
+            }
+        }
+    }
+
+    public render() {
+        const props = this.props;
+        return (
+            <View style={props.styles.container}>
+                <View style={props.styles.topContainer}>
+                    <View style={props.styles.mnemonicContainer}>
+                        {this.state.mnemonic.reduce((out: any, word: string, i: number) => {
+                            if (i % 4 === 0) {
+                                const line = this.state.mnemonic.slice(i, i + 4);
+                                out = [
+                                    ...out,
+                                    <View style={props.styles.mnemonicLine} key={i}>
+                                        {line.map((w, k) => (
+                                            <Text small key={k} style={props.styles.mnemonicWord}>
+                                                {i + k + 1}. {w}
+                                            </Text>
+                                        ))}
+                                    </View>
+                                ];
+                            }
+                            return out;
+                        }, [])}
+                    </View>
+                </View>
+                <View style={props.styles.bottomContainer}>
+                    <Button
+                        style={props.styles.bottomButton}
+                        primary
+                        onPress={() => {
+                            this.props.navigation.goBack(null);
+                        }}
+                    >
+                        {translate('App.labels.close')}
+                    </Button>
+                </View>
+
+                <PasswordModal
+                    buttonLabel={translate('Wallets.unveil')}
+                    infoText={translate('Wallets.unveilPasswordRequest')}
+                    visible={this.state.showPasswordModal}
+                    onReject={() => {
+                        this.props.navigation.goBack(null);
+                    }}
+                    onConfirm={password => {
```

Comment:
for this type of cases, i propose we use `@bind` decorator (https://www.npmjs.com/package/bind-decorator).
the code would look like this:
```ts
@bind
public async onEnterPassword(password) {...}

render() {
  return (...
    onConfirm={this.onEnterPassword}
  ...);
}
```

What do you say ?

Reply:
Cool. Added :)